### PR TITLE
GCS_MAVLink: count parameters in param thread

### PR DESCRIPTION
**Pull request by peterbarker**
This helps avoid counting parameters on the main thread,
avoiding long-loops

### CODE REVIEW

**Comment by tridge** [comment]
thanks, maybe remove the other call to register_io_process() ?

**Comment by peterbarker** [comment]
@tridge I did consider it, but decided to leave well-enough along; wasn't sure what else might be depending on tha tto be initialised and didn't want to rely on the thread to do that initialisation for it.

**Comment by peterbarker** [comment]
@tridge removed other call.

**Comment by OXINARF** [comment]
LGTM

File: libraries/GCS_MAVLink/GCS_Param.cpp [hunk 1]
```diff
@@ -375,6 +374,10 @@ void GCS_MAVLINK::param_io_timer(void)
 {
     struct pending_param_request req;
 
+    // this is mostly a no-op, but doing this here means we won't
+    // block the main thread counting parameters (~30ms on PH)
+    AP_Param::count_parameters();
```
Comment (magicrub) [inline libraries/GCS_MAVLink/GCS_Param.cpp]: if this can block ~30ms on PH, shouldn't the function get broken up?

Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Param.cpp]: its being called from the IO thread, which is the right place for slow calls, and its only 30ms once per change in parameter tree shape.

**Pull request merged**